certification/internal/engine/engine_test.go

```diff
@@ -1,10 +1,111 @@
 package engine
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http/httptest"
+	"net/url"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/random"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
 )
 
+var _ = Describe("Execute Checks tests", func() {
+	var src string
+	var engine CraneEngine
+	BeforeEach(func() {
+		// Set up a fake registry.
+		s := httptest.NewServer(registry.New())
+		DeferCleanup(func() {
+			s.Close()
+		})
+		u, err := url.Parse(s.URL)
+		Expect(err).ToNot(HaveOccurred())
+
+		src = fmt.Sprintf("%s/test/crane", u.Host)
+
+		// Expected values.
+		img, err := random.Image(1024, 5)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = crane.Push(img, src)
+		Expect(err).ToNot(HaveOccurred())
+
+		goodCheck := certification.NewGenericCheck(
+			"testcheck",
+			func(context.Context, certification.ImageReference) (bool, error) {
+				return true, nil
+			},
+			certification.Metadata{},
+			certification.HelpText{},
+		)
+
+		errorCheck := certification.NewGenericCheck(
+			"errorCheck",
+			func(context.Context, certification.ImageReference) (bool, error) {
+				return false, errors.New("errorCheck")
+			},
+			certification.Metadata{},
+			certification.HelpText{},
+		)
+
+		failedCheck := certification.NewGenericCheck(
+			"failedCheck",
+			func(context.Context, certification.ImageReference) (bool, error) {
+				return false, nil
+			},
+			certification.Metadata{},
+			certification.HelpText{},
+		)
+
+		engine = CraneEngine{
+			Image: src,
+			Checks: []certification.Check{
+				goodCheck,
+				errorCheck,
+				failedCheck,
+			},
+			IsBundle:  false,
+			IsScratch: false,
+		}
+	})
+	Context("Run the checks", func() {
+		ctx := context.TODO()
+		It("should succeed", func() {
+			err := engine.ExecuteChecks(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(engine.results.Passed).To(HaveLen(1))
+			Expect(engine.results.Failed).To(HaveLen(1))
+			Expect(engine.results.Errors).To(HaveLen(1))
+			Expect(engine.results.CertificationHash).To(BeEmpty())
+		})
+		Context("it is a bundle", func() {
+			JustBeforeEach(func() {
+				engine.IsBundle = true
+			})
```

Comment:
No change required with this comment. Just something I'm noting to commit to my own memory (and sharing if it helps).

This use of `JustBeforeEach` would be unnecessary because we only have a single `It` block below it. OTOH, if we had multiples, and they ran the chance of changing the value of `engine.IsBundle`, this would reset state for us.

I think this probably _could_ be restructured so that there are separate specs that both live within the top-level context having access to the configured `engine`, but this seems more than sufficient for this.

Reply:
You are correct. This is probably a misuse of JustBeforeEach. I can just set it right before calling the method under test. I am actually going to update this.